### PR TITLE
drivers base implementation

### PR DESCRIPTION
**Pull request by tonistiigi**
Starting to implement docker driver

### CODE REVIEW

**Comment by tonistiigi** [comment]
This is ready now. PTAL @tiborvass @AkihiroSuda

File: driver/manager.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+type InitConfig struct {
+	// This object needs updates to be generic for different drivers
+	Name           string
+	DockerAPI      dockerclient.APIClient
+	BuildkitConfig BuildkitConfig
+	Meta           map[string]interface{}
+}
+
+var drivers map[string]Factory
```
Comment (tiborvass) [inline driver/manager.go]: `var drivers = map[string]Factory{}`

Comment (tonistiigi) [reply driver/manager.go]: there is a nil check on line 34

Comment (tiborvass) [reply driver/manager.go]: Hence my suggestion, so you can get rid of that :)

File: driver/docker/driver.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+func (d *Driver) Client(ctx context.Context) (*client.Client, error) {
+	_, conn, err := d.exec(ctx, []string{"buildctl", "dial-stdio"})
+	if err != nil {
+		return nil, err
+	}
+
+	conn = demuxConn(conn)
+
+	return client.New(ctx, "", client.WithDialer(func(string, time.Duration) (net.Conn, error) {
```
Comment (AkihiroSuda) [inline driver/docker/driver.go]: Just curious, why not use `docker://...` address?

**Pull request merged**